storage/replicate_queue.go

```diff
@@ -75,21 +75,35 @@ func (rq replicateQueue) shouldQueue(now proto.Timestamp, repl *Replica) (should
 		return
 	}
 
-	return rq.needsReplication(zone, repl, repl.Desc())
+	delta := rq.replicaDelta(zone, repl, repl.Desc())
+	if delta == 0 {
+		if log.V(1) {
+			log.Infof("%s has the correct number of nodes", repl)
+		}
+		return false, 0
+	}
+	if delta > 0 {
+		if log.V(1) {
+			log.Infof("%s needs to add %d nodes", repl, delta)
+		}
+		// For ranges which need additional replicas, increase the priority
+		return true, float64(delta + 10)
+	}
+	if log.V(1) {
+		log.Infof("%s needs to remove %d nodes", repl, 0-delta)
+	}
+	// For ranges which have too many replicas, priority is absolute value of
+	// the delta.
+	return true, float64(0 - delta)
```

Comment:
I think this is splitting hairs at this point; if a Range has been designated as needing five replicas, it is probably considered more important range than one that has the default setting of 3, so maybe we should prioritize it anyway.